src/main/java/de/tum/cit/aet/artemis/programming/service/localci/distributedData/api/DistributedDataProvider.java

```diff
@@ -0,0 +1,71 @@
+package de.tum.cit.aet.artemis.programming.service.localci.distributedData.api;
+
+import java.util.Set;
+
+import de.tum.cit.aet.artemis.programming.service.localci.distributedData.api.map.DistributedMap;
+import de.tum.cit.aet.artemis.programming.service.localci.distributedData.api.queue.DistributedQueue;
+import de.tum.cit.aet.artemis.programming.service.localci.distributedData.api.topic.DistributedTopic;
+
+public interface DistributedDataProvider {
```

Comment:
Add a high level doc what the purpose / use case for this interface is. It's very generic (which I like!) but since requires a bit of explanation imo